tests/cpp_api/audio_tests/audio_tests.cpp

```diff
@@ -0,0 +1,208 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <unistd.h>
+
+#include <chrono>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+#include "rocal_api.h"
+
+using namespace std::chrono;
+
+bool verify_output(float *dst_ptr, long int frames, std::string case_name) {
+    bool pass_status = false;
+    // read data from golden outputs
+    const char *rocal_data_path = std::getenv("ROCAL_DATA_PATH");
+    if (strcmp(rocal_data_path, "") == 0) {
+        std::cout << "\n ROCAL_DATA_PATH env variable has not been set. ";
+        exit(0);
+    }
+
+    std::string ref_file_path = std::string(rocal_data_path) + "rocal_data/GoldenOutputsTensor/reference_outputs_audio/" + case_name + "_output.bin";
+    long int out_buffer_size = frames;
+    std::vector<float> ref_output(out_buffer_size);
+    std::fstream fin(ref_file_path, std::ios::in | std::ios::binary);
+    if (fin.is_open()) {
+        for (long int i = 0; i < out_buffer_size; i++) {
+            if (!fin.eof()) {
+                fin.read(reinterpret_cast<char *>(ref_output.data()), sizeof(float));
+            } else {
+                std::cout << "\nUnable to read all data from golden outputs\n";
+                return pass_status;
+            }
+        }
+    } else {
+        std::cout << "\nCould not open the reference output. Please check the path specified\n";
+        return pass_status;
+    }
+
+    int matched_indices = 0;
+    for (int j = 0; j < frames; j++) {
+        float ref_val, out_val;
+        ref_val = ref_output[j];
+        out_val = dst_ptr[j];
+        bool invalid_comparison = ((out_val == 0.0f) && (ref_val != 0.0f));
+        if (!invalid_comparison && abs(out_val - ref_val) < 1e-20)
+            matched_indices += 1;
+    }
+
+    std::cout << std::endl << "Results for Test case: " << std::endl;
+    if ((matched_indices == frames) && matched_indices != 0) {
+        pass_status = true;
+    }
+
+    return pass_status;
+}
+
+int test(int test_case, const char *path, int qa_mode, int downmix, int gpu);
+int main(int argc, const char **argv) {
```

Comment:
how does the user knows what each test_case corresponds to. We need to give a string parameter which can map to a test case so user arguments are more clear.

Reply:
If we accept a string from user, the user may pass the string in mixed cases, with or without underscore in the case of audio_decoder or for any other augmentation which would complicate the unit test. Hence similar to other unit tests we have accepted a number for the test case and internally we map to the string [here](https://github.com/swetha097/rocAL/blob/67cda83e168d31507ddd515be127fa770bc22b38/tests/cpp_api/audio_tests/audio_tests.py#L27), We will update the README with test case details